Examples incorporating alcotest-lwt

I'm not exactly sure how to use this package to test `lwt`, I tried to look at two other repos that depend on `alcotest-lwt` but I couldn't quite figure out how it should be used. Is there any chance there could be some simple examples on how best to incorporate `alcotest-lwt` into our tests?

The `'a` argument is what is passed to the test cases and you need to pass an extra arguments to the tests. Most of the time, it is indeed `()` and you have nothing to worry about.

However, if you want to use that parameter, you can do something like that:

```
let test_int i = Alcotest.(check int) "This is a nice integer" i 42

let int =
  let doc = "What is your prefered number?" in
  Cmdliner.Arg.(required & opt (some int) None & info ["n"] ~doc ~docv:"NUM")

let () =
  Alcotest.run_with_args "foo" int [
    "all", ["int", `Quick, test_int]
  ]
```

And:

```
$ _build/default/foo.exe test
foo.exe: required option -n is missing
$ _build/default/foo.exe test -n 42
Testing foo.
[OK]                all          0   int.
The full test results are available in `_build/_tests`.
Test Successful in 0.000s. 1 test run.
```

(note you must use a optional argument here, not a positional one otherwise alcotest will be confused). 

The switch is useful for `lwt` tests which need to manage resources, for instance when there is a global failure and you want to free up everything:

```
let free () = print_endline "freeing all resources"; Lwt.return ()

let test_lwt switch () =
  Lwt_switch.add_hook (Some switch) free;
  Lwt.async (fun () -> failwith "All is broken");
  Lwt_unix.sleep 10.

let () =
  Alcotest.run "foo" [
    "all", [
      Alcotest_lwt.test_case "one" `Quick test_lwt
    ]
  ]
```

will display:

```
$ _build/default/foo.exe
Testing foo.
[ERROR]             all          0   one.
-- all.000 [one.] Failed --
in _build/_tests/all.000.output:
freeing all resources
[failure] All is broken
```

--- a/README.md
+++ b/README.md
@@ -59,3 +59,87 @@ Test Successful in 0.001s. 2 tests run.
 
 See the [examples](https://github.com/mirage/alcotest/tree/master/examples)
 folder for more examples.
+
+### Passing custom options to the tests
+
+In most cases, the base tests are `unit -> unit` functions. However,
+it is also possible to pass an extra option to all the test functions
+by using `'a -> unit`, where `'a` is the type of the extra parameter.
+
+In order to do this, you need to specify how this extra parameter is
+read on the command-line, by providing a [Cmdliner term for
+command-line
+arguments](http://erratique.ch/software/cmdliner/doc/Cmdliner.Term.html)
+which explains how to parse and serialize values of type `'a` (*note:* do not
+use positional arguments, only optional arguments are supported).
+
+For instance:
+
+```ocaml
+let test_nice i = Alcotest.(check int) "Is it a nice integer?" i 42
+
+let int =
+  let doc = "What is your prefered number?" in
+  Cmdliner.Arg.(required & opt (some int) None & info ["n"] ~doc ~docv:"NUM")
+
+let () =
+  Alcotest.run_with_args "foo" int [
+    "all", ["nice", `Quick, test_nice]
+  ]
+```
+
+Will generate `test.exe` such that:
+
+```
+$ test.exe test
+test.exe: required option -n is missing
+
+$ test.exe test -n 42
+Testing foo.
+[OK]                all          0   int.
+```
+
+### Lwt
+
+Alcotest provides an `Alcotest_lwt` module that you could use to wrap
+Lwt test cases. The basic idea is that instead of providing a test
+function in the form `unit -> unit`, you provide one with the type
+`unit -> unit Lwt.t` and alcotest-lwt calls `Lwt_main.run` for you.
+
+However, there are a couple of extra features:
+
+- If an async exception occurs, it will cancel your test case for you
+  and fail it (rather than exiting the process).
+
+- You get given a switch, which will be turned off when the test case
+  finishes (or fails). You can use that to free up any resources.
+
+For instance:
+
+```ocaml
+let free () = print_endline "freeing all resources"; Lwt.return ()
+
+let test_lwt switch () =
+  Lwt_switch.add_hook (Some switch) free;
+  Lwt.async (fun () -> failwith "All is broken");
+  Lwt_unix.sleep 10.
+
+let () =
+  Alcotest.run "foo" [
+    "all", [
+      Alcotest_lwt.test_case "one" `Quick test_lwt
+    ]
+  ]
+```
+
+Will generate:
+
+```
+$ test.exe
+Testing foo.
+[ERROR]             all          0   one.
+-- all.000 [one.] Failed --
+in _build/_tests/all.000.output:
+freeing all resources
+[failure] All is broken
+```

--- a/README.md
+++ b/README.md
@@ -60,6 +60,19 @@ Test Successful in 0.001s. 2 tests run.
 See the [examples](https://github.com/mirage/alcotest/tree/master/examples)
 folder for more examples.
 
+### Quick and Slow tests
+
+In general you should be using `` `Quick`` tests; `` `Slow`` tests
+tests are useful when you are defined stress tests, that do not need
+to run on every `make test` but that you will typically run before
+a release. This mechanism is a way to mark some tests as optional; they
+will be not triggered when you pass -q on the command line, e.g.:
+
+```
+$ ./test.exe -q # run only the quick tests
+$ ./test.exe    # run quick and slow tests
+```
+
 ### Passing custom options to the tests
 
 In most cases, the base tests are `unit -> unit` functions. However,

